pkg/controller/mongodb/replica_set_controller.go

```diff
@@ -551,6 +551,16 @@ func (r ReplicaSetReconciler) validateUpdate(mdb mdbv1.MongoDB) error {
 	return validation.Validate(prevSpec, mdb.Spec)
 }
 
+func getCustomRolesModification(mdb mdbv1.MongoDB) (automationconfig.Modification, error) {
+	if mdb.Spec.Roles == nil {
```

Comment:
what do you think about moving `mdb.Spec.Roles` to `mdb.Spec.Security.Authentication.Roles`? I think it makes more sense to group these roles with the other auth settings.

Reply:
Done!